tfhe/src/boolean/public_key/standard.rs

```diff
@@ -61,6 +61,55 @@ impl PublicKey {
     pub fn new(client_key: &ClientKey) -> Self {
         BooleanEngine::with_thread_local_mut(|engine| engine.create_public_key(client_key))
     }
+
+    /// Deconstruct a [`PublicKey`] into its constituants.
+    pub fn into_raw_parts(self) -> (LwePublicKeyOwned<u32>, BooleanParameters) {
+        let Self {
+            lwe_public_key,
+            parameters,
+        } = self;
+
+        (lwe_public_key, parameters)
+    }
+
+    /// Construct a [`PublicKey`] from its constituants.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the constituants are not compatible with each others.
+    pub fn from_raw_parts(
+        lwe_public_key: LwePublicKeyOwned<u32>,
+        parameters: BooleanParameters,
+    ) -> Self {
+        let ciphertext_lwe_dimension = match parameters.encryption_key_choice {
+            crate::core_crypto::commons::parameters::EncryptionKeyChoice::Big => parameters
+                .glwe_dimension
+                .to_equivalent_lwe_dimension(parameters.polynomial_size),
+            crate::core_crypto::commons::parameters::EncryptionKeyChoice::Small => {
+                parameters.lwe_dimension
+            }
+        };
```

Comment:
This could have been a Boolean parameters method

Reply:
yep you found the follow up